### PR TITLE
ensure unique buildingId and latest readin in sql trigger

### PR DESCRIPTION
**Pull request by hanzzhang**
connects to #96

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hanzzhang__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by lbrader** [comment]
This should also address #67.  testing

File: provision/SqlDatabase/CreateSqlDatabase_Schema.sql [hunk 1]
```diff
@@ -22,11 +22,21 @@ ON [dbo].[BuildingTemperature]
...
+			GROUP BY BuildingId 
+		) AS UniqueBuildingLatestedData
+		ON AllData.BuildingId = UniqueBuildingLatestedData.BuildingId AND AllData.LastObservedTime = UniqueBuildingLatestedData.LastObservedTime 
+	)
 	MERGE dbo.BuildingTemperature AS T  
-	USING INSERTED AS S 
-	ON T.BuildingId = S.BuildingId 
-	WHEN MATCHED THEN  
-		UPDATE SET T.LastObservedTime = S.LastObservedTime, T.Temperature = S.Temperature 
+	USING DistinctBuidlingBatch AS S
```
Comment (mj1856) [inline provision/SqlDatabase/CreateSqlDatabase_Schema.sql]: Typo - Building, not Buidling

File: provision/SqlDatabase/CreateSqlDatabase_Schema.sql [hunk 2]
```diff
@@ -22,11 +22,21 @@ ON [dbo].[BuildingTemperature]
 INSTEAD OF INSERT 
 AS
 BEGIN 
+	WITH DistinctBuidlingBatch AS
```
Comment (mj1856) [inline provision/SqlDatabase/CreateSqlDatabase_Schema.sql]: Typo - Building, not Buidling

**Pull request merged**